### DISCUSSION TITLE
How to use PCM to monitor UPI bandwidth?

### DISCUSSION THREAD

**Discussion by jasonneverstop** [opcm/pcm#555] [General]
How to use PCM to monitor UPI bandwidth?

**Comment by ogbrugge** [comment]
Hi,

UPI bandwidth can be measured with the pcm binary, just run it.
If you do not want the core metrics, run pcm with -nc
If you do not care about the system aggregation run pcm with -nsys
You can combine these options to only see socket level information
You can add -csv=filename to output to a csv file.

This info is all available in the pcm --help output.

Hope this helps.

**Reply by jasonneverstop** [reply]
thank you.
As shown in the image, is the local access ratio of memory for SKT0 91%? The remote access ratio is calculated as 1-91% = 9%？
![image](https://github.com/intel/pcm/assets/68673502/322f0cfd-e69d-42af-b91b-807a34bab92c)

**Reply by rdementi** [reply]
yes, correct

**Reply by hangfu-guo** [reply]
Hi @rdementi  As to the LOCAL value,  when there's no workload running on  the machine, I can see that the LOCAL of both nodes is around 60%~70%, is this normal?  BTW,  for Xeon MAX(with HBM), what's the definition of LOCAL here, HBM or DDR close to the node(HBM Flat mode)?

**Comment by rdementi** [comment]
Hi,

yes, absolutely this is normal. Also when you are not running any heavy workload your system executes CPU instructions in background (e.g. OS kernel, monitoring tools, etc) which can use both local and remote memory. The definition of local is with respect to the socket. Events in these categories are used to compute the metric:
https://github.com/intel/perfmon/blob/47778caa4d75ddce5e558e95682561289b370579/SPR/events/sapphirerapids_uncore.json#L2107
https://github.com/intel/perfmon/blob/47778caa4d75ddce5e558e95682561289b370579/SPR/events/sapphirerapids_uncore.json#L2125

e.g. accessing HBM or DDR on the same socket does not make any difference. It is all local.
